File: vb/OperadoresLogicosVB/OperadoresLogicosVB/Form1.vb
Public Class frmOperadoresLogicos

    Private Sub frmOperadoresLogicos_Load(sender As Object, e As EventArgs) Handles MyBase.Load

        cmbArit.SelectedIndex = 0
        cmbLog.SelectedIndex = 1
        cmbLog1.SelectedIndex = 0
        cmbLog2.SelectedIndex = 0
        cmbRel.SelectedIndex = 0

    End Sub



    Private Sub txtArit1_KeyPress(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyPressEventArgs) Handles txtArit1.KeyPress

        If txtArit1.Text = String.Empty Then

            e.Handled = Not Char.IsDigit(e.KeyChar) And Not Char.IsControl(e.KeyChar) And Not e.KeyChar = "." And Not e.KeyChar = "-"

        ElseIf Not txtArit1.Text.Contains(".") Then

            e.Handled = Not Char.IsDigit(e.KeyChar) And Not Char.IsControl(e.KeyChar) And Not e.KeyChar = "."

        ElseIf Not txtArit1.Text.Contains("-") Then

            e.Handled = Not Char.IsDigit(e.KeyChar) And Not Char.IsControl(e.KeyChar) And Not e.KeyChar = "-"

        ElseIf txtArit1.Text.Contains(".") And txtArit1.Text.Contains("-") Then

            e.Handled = Not Char.IsDigit(e.KeyChar) And Not Char.IsControl(e.KeyChar)

        End If

    End Sub

    Private Sub txtArit2_KeyPress(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyPressEventArgs) Handles txtArit2.KeyPress

        If txtArit2.Text = String.Empty Then

            e.Handled = Not Char.IsDigit(e.KeyChar) And Not Char.IsControl(e.KeyChar) And Not e.KeyChar = "." And Not e.KeyChar = "-"

        ElseIf Not txtArit2.Text.Contains(".") Then

            e.Handled = Not Char.IsDigit(e.KeyChar) And Not Char.IsControl(e.KeyChar) And Not e.KeyChar = "."

        ElseIf Not txtArit2.Text.Contains("-") Then

            e.Handled = Not Char.IsDigit(e.KeyChar) And Not Char.IsControl(e.KeyChar) And Not e.KeyChar = "-"

        ElseIf txtArit2.Text.Contains(".") And txtArit2.Text.Contains("-") Then

            e.Handled = Not Char.IsDigit(e.KeyChar) And Not Char.IsControl(e.KeyChar)

        End If

    End Sub







    'Aritmeticos
    Public Sub cmbArit_SelectedIndexChanged(sender As Object, e As EventArgs) Handles cmbArit.SelectedIndexChanged

        If ((txtArit1.Text IsNot String.Empty) And (txtArit2.Text IsNot String.Empty)) Then

            If txtArit1.Text Like "*.*-*" Or txtArit1.Text Like "." Or txtArit1.Text Like "-" Or txtArit1.Text Like "-." Or txtArit2.Text Like "*.-*" Or txtArit2.Text Like "." Or txtArit2.Text Like "-" Or txtArit2.Text Like "-." Then

                MsgBox("Valor invalido")
                Exit Sub

            End If

            If cmbArit.SelectedItem = "+" Then

                MsgBox(Convert.ToDouble(txtArit1.Text) + Convert.ToDouble(txtArit2.Text))

            ElseIf cmbArit.SelectedItem = "-" Then

                MsgBox(Convert.ToDouble(txtArit1.Text) - Convert.ToDouble(txtArit2.Text))

            ElseIf cmbArit.SelectedItem = "*" Then

                MsgBox(Convert.ToDouble(txtArit1.Text) * Convert.ToDouble(txtArit2.Text))

            ElseIf cmbArit.SelectedItem = "/" Then

                MsgBox(Convert.ToDouble(txtArit1.Text) / Convert.ToDouble(txtArit2.Text))

            ElseIf cmbArit.SelectedItem = "\" Then

                MsgBox(Convert.ToDouble(txtArit1.Text) \ Convert.ToDouble(txtArit2.Text))

            ElseIf cmbArit.SelectedItem = "Mod" Then

                MsgBox(Convert.ToDouble(txtArit1.Text) Mod Convert.ToDouble(txtArit2.Text))

            ElseIf cmbArit.SelectedItem = "^" Then

                MsgBox(Convert.ToDouble(txtArit1.Text) ^ Convert.ToDouble(txtArit2.Text))

            End If

        End If

    End Sub





    'Logicos
    Private Sub cmbLog_SelectedIndexChanged(sender As Object, e As EventArgs) Handles cmbLog.SelectedIndexChanged

        cmbLog2.Enabled = True

        If cmbLog1.Text IsNot String.Empty Then

            If cmbLog.SelectedItem = "Not" Then

                cmbLog2.Enabled = False
                MsgBox(Not (Convert.ToBoolean(cmbLog1.SelectedIndex)))

            End If

        End If

        If ((cmbLog1.Text IsNot String.Empty) And (cmbLog2.Text IsNot String.Empty)) Then

            If cmbLog.SelectedItem = "And" Then

                MsgBox(Convert.ToBoolean(cmbLog1.SelectedIndex) And Convert.ToBoolean(cmbLog2.SelectedIndex))

            ElseIf cmbLog.SelectedItem = "Or" Then

                MsgBox(Convert.ToBoolean(cmbLog1.SelectedIndex) Or Convert.ToBoolean(cmbLog2.SelectedIndex))

            ElseIf cmbLog.SelectedItem = "Xor" Then

                MsgBox(Convert.ToBoolean(cmbLog1.SelectedIndex) Xor Convert.ToBoolean(cmbLog2.SelectedIndex))

            End If

        End If

    End Sub








    'Relacionales
    Private Sub cmbRel_SelectedIndexChanged(sender As Object, e As EventArgs) Handles cmbRel.SelectedIndexChanged

        If ((txtRel1.Text IsNot String.Empty) And (txtRel2.Text IsNot String.Empty)) Then


            If cmbRel.SelectedItem = ">" Then

                MsgBox(txtRel1.Text > txtRel2.Text)

            ElseIf cmbRel.SelectedItem = "<" Then

                MsgBox(txtRel1.Text < txtRel2.Text)

            ElseIf cmbRel.SelectedItem = "<=" Then

                MsgBox(txtRel1.Text <= txtRel2.Text)

            ElseIf cmbRel.SelectedItem = ">=" Then

                MsgBox(txtRel1.Text >= txtRel2.Text)

            ElseIf cmbRel.SelectedItem = "=" Then

                MsgBox(txtRel1.Text = txtRel2.Text)

            ElseIf cmbRel.SelectedItem = "<>" Then

                MsgBox(txtRel1.Text <> txtRel2.Text)

            End If


        End If

    End Sub


End Class

File: cs/OperadoresCS/OperadoresCS/Form1.cs
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace OperadoresCS
{
	public partial class frmOperadoresCS : Form
	{
		public frmOperadoresCS()
		{
			InitializeComponent();
		}

		private void frmOperadoresCS_Load(object sender, EventArgs e)
		{

			cmbArit.SelectedIndex = 0;
			cmbLog.SelectedIndex = 1;
			cmbLog1.SelectedIndex = 0;
			cmbLog2.SelectedIndex = 0;
			cmbRel.SelectedIndex = 0;

		}

		private void txtArit1_KeyPress(object sender, KeyPressEventArgs e)
		{
			if (string.IsNullOrEmpty(txtArit1.Text))
			{

				e.Handled = !char.IsDigit(e.KeyChar) && !char.IsControl(e.KeyChar) && !(e.KeyChar.Equals(".")) && !(e.KeyChar.Equals("-"));
			}
			else if (!txtArit1.Text.Contains("."))
			{

				e.Handled = !char.IsDigit(e.KeyChar) && !char.IsControl(e.KeyChar) && !(e.KeyChar.Equals("."));

			}
			else if (!txtArit1.Text.Contains("-"))
			{

				e.Handled = !char.IsDigit(e.KeyChar) && !char.IsControl(e.KeyChar) && !(e.KeyChar.Equals("-"));

			}
			else if (txtArit1.Text.Contains(".") & txtArit1.Text.Contains("-"))
			{

				e.Handled = !char.IsDigit(e.KeyChar) && !char.IsControl(e.KeyChar);

			}

		}

		private void txtArit2_KeyPress(object sender, KeyPressEventArgs e)
		{

			if (string.IsNullOrEmpty(txtArit2.Text))
			{

				e.Handled = !char.IsDigit(e.KeyChar) && !char.IsControl(e.KeyChar) && !(e.KeyChar.Equals(".")) && !(e.KeyChar.Equals("-"));
			}
			else if (!txtArit2.Text.Contains("."))
			{

				e.Handled = !char.IsDigit(e.KeyChar) && !char.IsControl(e.KeyChar) && !(e.KeyChar.Equals("."));

			}
			else if (!txtArit2.Text.Contains("-"))
			{

				e.Handled = !char.IsDigit(e.KeyChar) && !char.IsControl(e.KeyChar) && !(e.KeyChar.Equals("-"));

			}
			else if (txtArit2.Text.Contains(".") & txtArit1.Text.Contains("-"))
			{

				e.Handled = !char.IsDigit(e.KeyChar) && !char.IsControl(e.KeyChar);

			}

		}









		//Aritmeticos
		private void cmbArit_SelectedIndexChanged(object sender, EventArgs e)
		{

			if ((!string.IsNullOrEmpty(txtArit1.Text)) && (!string.IsNullOrEmpty(txtArit2.Text)))
			{

				if (LikeOperator.LikeString(txtArit1.Text, "*.*-*", CompareMethod.Binary) | LikeOperator.LikeString(txtArit1.Text, ".", CompareMethod.Binary) | LikeOperator.LikeString(txtArit1.Text, "-", CompareMethod.Binary) | LikeOperator.LikeString(txtArit1.Text, "-.", CompareMethod.Binary) | LikeOperator.LikeString(txtArit2.Text, "*.-*", CompareMethod.Binary) | LikeOperator.LikeString(txtArit2.Text, ".", CompareMethod.Binary) | LikeOperator.LikeString(txtArit2.Text, "-", CompareMethod.Binary) | LikeOperator.LikeString(txtArit2.Text, "-.", CompareMethod.Binary))
				{
					MessageBox.Show("Valor invalido");
					return;
				}

				




				if (cmbArit.SelectedItem.Equals("+"))
				{

					MessageBox.Show(Convert.ToString(Convert.ToDouble(txtArit1.Text) + Convert.ToDouble(txtArit2.Text)));

				}
				else if (cmbArit.SelectedItem.Equals("-"))
				{

					MessageBox.Show(Convert.ToString(Convert.ToDouble(txtArit1.Text) - Convert.ToDouble(txtArit2.Text)));

				}
				else if (cmbArit.SelectedItem.Equals("*"))
				{

					MessageBox.Show(Convert.ToString(Convert.ToDouble(txtArit1.Text) * Convert.ToDouble(txtArit2.Text)));

				}
				else if (cmbArit.SelectedItem.Equals("/"))
				{

					MessageBox.Show(Convert.ToString(Convert.ToDouble(txtArit1.Text) / Convert.ToDouble(txtArit2.Text)));

				}
				else if (cmbArit.SelectedItem.Equals("\\"))
				{

					MessageBox.Show(Convert.ToString(Convert.ToInt32(Convert.ToDouble(txtArit1.Text)) / Convert.ToInt32(Convert.ToDouble(txtArit2.Text))));

				}
				else if (cmbArit.SelectedItem.Equals("Mod"))
				{

					MessageBox.Show(Convert.ToString(Convert.ToDouble(txtArit1.Text) % Convert.ToDouble(txtArit2.Text)));

				}
				else if (cmbArit.SelectedItem.Equals("^"))
				{

					MessageBox.Show(Convert.ToString(Math.Pow(Convert.ToDouble(txtArit1.Text), Convert.ToDouble(txtArit2.Text))));

				}
			}

		}







		//Logicos

		private void cmbLog_SelectedIndexChanged(object sender, EventArgs e)
		{

			

			if (!string.IsNullOrEmpty(cmbLog1.Text))
			{

				if (cmbLog.SelectedItem.Equals( "Not"))
				{

					cmbLog2.Enabled = false;
					MessageBox.Show(Convert.ToString(!(Convert.ToBoolean(cmbLog1.SelectedIndex))));

				}
				else
				{
					cmbLog2.Enabled = true;
				}

			}


			if ((!string.IsNullOrEmpty(cmbLog1.Text)) && (!string.IsNullOrEmpty(cmbLog2.Text)))
			{
				
				if (cmbLog.SelectedItem.Equals( "And"))
				{

					MessageBox.Show(Convert.ToString(Convert.ToBoolean(cmbLog1.SelectedIndex) & Convert.ToBoolean(cmbLog2.SelectedIndex)));

				}
				else if (cmbLog.SelectedItem.Equals("Or"))
				{

					MessageBox.Show(Convert.ToString(Convert.ToBoolean(cmbLog1.SelectedIndex) | Convert.ToBoolean(cmbLog2.SelectedIndex)));

				}
				else if (cmbLog.SelectedItem.Equals("Xor"))
				{

					MessageBox.Show(Convert.ToString(Convert.ToBoolean(cmbLog1.SelectedIndex) ^ Convert.ToBoolean(cmbLog2.SelectedIndex)));

				}


			}



		}





		//Relacionales
		private void cmbRel_SelectedIndexChanged(object sender, EventArgs e)
		{

			if (txtRel1.Text != string.Empty & txtRel2.Text != string.Empty)
			{
				if (Conversions.ToBoolean(Operators.ConditionalCompareObjectEqual(cmbRel.SelectedItem, ">", false)))
				{
					MessageBox.Show(Convert.ToString(txtRel1.Text.Length > txtRel2.Text.Length));
				}
				else if (Conversions.ToBoolean(Operators.ConditionalCompareObjectEqual(cmbRel.SelectedItem, "<", false)))
				{
					MessageBox.Show(Convert.ToString(txtRel1.Text.Length < txtRel2.Text.Length));
				}
				else if (Conversions.ToBoolean(Operators.ConditionalCompareObjectEqual(cmbRel.SelectedItem, "<=", false)))
				{
					MessageBox.Show(Convert.ToString(txtRel1.Text.Length <= txtRel2.Text.Length));
				}
				else if (Conversions.ToBoolean(Operators.ConditionalCompareObjectEqual(cmbRel.SelectedItem, ">=", false)))
				{
					MessageBox.Show(Convert.ToString(txtRel1.Text.Length >= txtRel2.Text.Length));
				}
				else if (Conversions.ToBoolean(Operators.ConditionalCompareObjectEqual(cmbRel.SelectedItem, "==", false)))
				{
					MessageBox.Show(Convert.ToString((txtRel1.Text) == (txtRel2.Text)));
				}
				else if (Conversions.ToBoolean(Operators.ConditionalCompareObjectEqual(cmbRel.SelectedItem, "!=", false)))
				{
					MessageBox.Show(Convert.ToString((txtRel1.Text) != (txtRel2.Text)));
				}
			}


		}

















	}
	}
